ow_plexil/src/plans/IdentifySampleTarget.plp

```diff
@@ -2,31 +2,47 @@
 // Research and Simulation can be found in README.md in the root directory of
 // this repository.
 
-// This is a simplified and stubbed version of a procedure that should
-// "interrogate" the terrain to find a good sampling location.  Instead, we
-// choose an arbitrary location and probe for ground position there.
+//Takes pictures of terrain using stereocamera in front of lander and sends them to action server for processing
+//Points of interest are determined by FilterType: "Brown" finds brown areas, "Dark" finds dark areas
+//It chooses the best sample point and then uses guarded move to find the exact ground position of that spot
 
 #include "plan-interface.h"
 
+LibraryAction PanAndShoot (In Real PanAngle);
+
 IdentifySampleTarget:
 {
   InOut Real X, Y, GroundPos;
   InOut Boolean Parallel;
-
+  In String FilterType;
+  Real result[3];
   Post Lookup (GroundFound);
 
-  // Guarded move is at attempt to find the ground position.  The chosen start
-  // position (X/Y/Z) is arbitrary, as is the search direction (vertical).
-  LibraryCall GuardedMove (X = 2.0, Y = 0, Z = 0.3,
-                           DirX = 0, DirY = 0, DirZ = 1,
-                           SearchDistance = 0.5);
-  if (Lookup (GroundFound)) GroundPos = Lookup (GroundPosition);
-  else log_warning ("GuardedMove failed to find ground.");
-  endif
-
-  // The chosen X/Y sampling location, as well as choice for Parallel trench
-  // direction are stubbed here; these choices are arbitrary.
-  X = 1.65;
-  Y = 0;
-  Parallel = true;
+  // Take pictures in front of the lander for processing in identify_sample_location
+  // An alternative would be to create a new takepicture command that waits for the pointcloud or modifying the current one
+  LibraryCall Tilt(Degrees = 50);
+  LibraryCall PanAndShoot(PanAngle = 65);
+  LibraryCall PanAndShoot(PanAngle = 70);
+  LibraryCall PanAndShoot(PanAngle = 75);
```

Comment:
I would recommend you define the center position (expressed in angles) and then use +/- for the extra shots. i.e:
```suggestion
  c_tilt = 50;
  c_pan = 70;
  delta = 5;
  LibraryCall Tilt(Degrees = c_tilt);
  LibraryCall PanAndShoot(PanAngle = c_pan - delta);
  LibraryCall PanAndShoot(PanAngle = c_pan);
  LibraryCall PanAndShoot(PanAngle = c_pan + delta);
```